crates/typst-cli/src/query.rs

```diff
@@ -0,0 +1,120 @@
+use std::time::Instant;
+
+use comemo::Track;
+use serde::Serialize;
+use typst::diag::{bail, StrResult};
+use typst::eval::{eval_string, EvalMode, Tracer};
+use typst::model::{Introspector, Selector};
+use typst::World;
+use typst_library::meta::ProvideElem;
+use typst_library::prelude::*;
+
+use crate::args::{OutputFormat, QueryCommand};
+use crate::compile::print_diagnostics;
+use crate::set_failed;
+use crate::world::SystemWorld;
+
+/// Execute a query command.
+pub fn query(command: QueryCommand) -> StrResult<()> {
+    let mut world = SystemWorld::new(&command.common)?;
+    tracing::info!("Starting querying");
+
+    let start = Instant::now();
+    // Reset everything and ensure that the main file is present.
+    world.reset();
+    world.source(world.main()).map_err(|err| err.to_string())?;
+
+    let mut tracer = Tracer::default();
+    let result = typst::compile(&world, &mut tracer);
+    let duration = start.elapsed();
+    let warnings = tracer.warnings();
+
+    match result {
+        // Print metadata
+        Ok(document) => {
+            query_and_format(&document, &command, &world)?;
+            tracing::info!("Processing succeeded in {duration:?}");
```

Comment:
I think the query command is a prime example of when user messages should be printed to stderr, and not stdout, as stdout will contain the data which external programs will attempt to parse. As I see it, we should ensure anything that isn't data here should go straight to stderr, when that isn't already the case.

I believe it'd be appropriate to do the same for the Err() case. (The return code of the command should distinguish whether or not the query was successful.)

Reply:
you're right, i'll have a look

Reply:
I would expect tracing outputs to print to stderr anyway. Isn't that the default? Currently I don't see those lines in my terminal at all. Can anyone else check with the latest main branch?

Reply:
If that's the case then that's OK; just wanted to make sure we're certain of this. Definitely might be a good idea to test it a bit. :)

Reply:
I will leave this conversation unresolved to remind us. But currently I do not see any of the tracing output, so I cannot test. Has the handling of tracing output changed lately?